internal/webhook/shared/helpers.go

```diff
@@ -0,0 +1,71 @@
+package shared
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// NewLogConstructor returns a log constructor function that adds the webhook name to the logger context for each admission request.
+func NewLogConstructor(name string) func(logr.Logger, *admission.Request) logr.Logger {
+	return func(_ logr.Logger, req *admission.Request) logr.Logger {
+		base := ctrl.Log
+		l := admission.DefaultLogConstructor(base, req)
+		return l.WithValues("webhook", name)
+	}
+}
+
+// CountObjects returns the number of objects of the given GroupVersionKind in the cluster using the provided client.
+// It lists all objects of the specified GVK and returns their count, or an error if the list operation fails.
+func CountObjects(ctx context.Context, cli client.Client, gvk schema.GroupVersionKind) (int, error) {
```

Comment:
Since this is a generic function, I'd recommend to pass also the `ListOption`  vararg so the function ca be reused or tweaked (i.e. set pagination, limits)